### PR TITLE
Refactor duplicated code in repo handlers

### PR DESCRIPTION
**Pull request by ethantkoenig**
Move duplicated code in `RepoIDAssignment` and `RepoAssignment` handlers to a helper function. Also removed an unused var and fix a typo

### CODE REVIEW

**Comment by daviian** [comment]
LGTM

File: modules/context/repo.go [hunk 1]
```diff
@@ -174,13 +174,54 @@ func RedirectToRepo(ctx *Context, redirectRepoID int64) {
...
+		ctx.Repo.AccessMode = mode
+	}
+
+	// Check access.
+	if ctx.Repo.AccessMode == models.AccessModeNone {
+		if ctx.Query("go-get") == "1" {
+			EarlyResponseForGoGetMeta(ctx)
+			return
+		}
+		ctx.Error(404, "no access right")
```
Comment (daviian) [inline modules/context/repo.go]: Please change to `ctx.Handle(404, "no access right", nil)`

**Review by daviian — CHANGES_REQUESTED** [review]

File: modules/context/repo.go (hunk 1 above)
Comment (ethantkoenig) [reply modules/context/repo.go]: Done

**Review by daviian — APPROVED** [review]

**Comment by lafriks** [comment]
LGTM

**Pull request merged**